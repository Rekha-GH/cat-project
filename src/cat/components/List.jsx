import React from "react";
import { useSelector } from "react-redux";
import { Grid } from "../../common/components";
import { getAll } from "../model/ImageModel";
import CatCard from "./CatCard";
import { getIdList } from "../Store/CatSelector";
import ErrorMessage from "../../common/components/ErrorMessage";

const List = React.memo(() => {
  const list = useSelector(getIdList);
  React.useEffect(() => {
    getAll();
  }, []);

  return (
    <>
      <ErrorMessage />
      <Grid>
        {list.map((x) => (
          <CatCard key={x} id={x} />
        ))}
      </Grid>
    </>
  );
});

export default List;
