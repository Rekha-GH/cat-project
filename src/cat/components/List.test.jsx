import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import * as ReactRedux from "react-redux";
import List from "./List";

Enzyme.configure({ adapter: new Adapter() });

describe("List Test", () => {
  const useSelectorMock = jest.spyOn(ReactRedux, "useSelector");
  jest.spyOn(ReactRedux, "useDispatch");

  test("List Call Correctly when list is not available", () => {
    useSelectorMock.mockReturnValue([]);
    const wrapper = shallow(<List />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("List Call Correctly when list is available", () => {
    useSelectorMock.mockReturnValueOnce([1, 2, 3]);
    const wrapper = shallow(<List />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
