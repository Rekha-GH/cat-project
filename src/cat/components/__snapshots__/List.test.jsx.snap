// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Test List Call Correctly when list is available 1`] = `
<Fragment>
  <Memo() />
  <Memo()>
    <Memo()
      id={1}
      key="1"
    />
    <Memo()
      id={2}
      key="2"
    />
    <Memo()
      id={3}
      key="3"
    />
  </Memo()>
</Fragment>
`;

exports[`List Test List Call Correctly when list is not available 1`] = `
<Fragment>
  <Memo() />
  <Memo() />
</Fragment>
`;
